sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/ReleaseDownloadListener.java

```diff
@@ -127,29 +105,30 @@ public boolean onComplete(@NonNull Uri localUri) {
              */
             AppCenterLog.info(LOG_TAG, "Show install UI for " + localUri);
             mContext.startActivity(intent);
-            Distribute distribute = Distribute.getInstance();
-            if (mReleaseDetails.isMandatoryUpdate()) {
-                distribute.setInstalling(mReleaseDetails);
-            } else {
-                distribute.completeWorkflow(mReleaseDetails);
-            }
-            storeReleaseDetails(mReleaseDetails);
+            Distribute.getInstance().setInstalling(mReleaseDetails);
         }
         return true;
     }
 
+    @WorkerThread
     @Override
     public void onError(@NonNull String errorMessage) {
         AppCenterLog.error(LOG_TAG, errorMessage);
-        Toast.makeText(mContext, R.string.appcenter_distribute_downloading_error, Toast.LENGTH_SHORT).show();
+        HandlerUtils.runOnUiThread(new Runnable() {
+
+            @Override
+            public void run() {
+                Toast.makeText(mContext, R.string.appcenter_distribute_downloading_error, Toast.LENGTH_SHORT).show();
```

Comment:
Was that necessary to post? I'm surprised toast was forbidden from worker thread.

Reply:
https://stackoverflow.com/questions/3875184

Reply:
Good catch.